### PR TITLE
Add docs to the time and fcntl modules

### PR DESCRIPTION
**Pull request by asomers**
Also, deprecate FlockArg::UnlockNonblock.  That combination of flags doesn't make sense.  And indeed, reviewing the kernel source code shows that the kernel will ignore LOCK_NB when LOCK_UN is set.

### CODE REVIEW

File: src/fcntl.rs [hunk 1]
```diff
@@ -45,15 +46,28 @@ pub use self::posix_fadvise::{posix_fadvise, PosixFadviseAdvice};
...
+    /// Flags that control how the various *at syscalls behave.
     #[cfg_attr(docsrs, doc(cfg(any(feature = "fs", feature = "process"))))]
     pub struct AtFlags: c_int {
+        /// Used with [`unlink`](crate::unistd::unlink) to indicate that the path to remove is a
+        /// directory, and not a regular file.
         AT_REMOVEDIR;
+        /// Used with [`linkat`](crate::unistd::linkat`) to create a link to a symbolic link's
+        /// target, instead of to the symbolic link itself.
         AT_SYMLINK_FOLLOW;
+        /// Used with [`linkat`](crate::unistd::linkat`) to create a link to a symbolic link
```
Comment (SteveLauC) [inline src/fcntl.rs:58]: `linkat()` does not support this flag as it is the default behavior per the [POSIX standard](https://pubs.opengroup.org/onlinepubs/9699919799/functions/link.html)

File: src/fcntl.rs [hunk 2]
```diff
@@ -45,15 +46,28 @@ pub use self::posix_fadvise::{posix_fadvise, PosixFadviseAdvice};
 #[cfg(not(target_os = "redox"))]
 #[cfg(any(feature = "fs", feature = "process", feature = "user"))]
 libc_bitflags! {
+    /// Flags that control how the various *at syscalls behave.
     #[cfg_attr(docsrs, doc(cfg(any(feature = "fs", feature = "process"))))]
     pub struct AtFlags: c_int {
+        /// Used with [`unlink`](crate::unistd::unlink) to indicate that the path to remove is a
```
Comment (SteveLauC) [inline src/fcntl.rs:52]: It should be `unlinkat()`:

```suggestion
        /// Used with [`unlinkat`](crate::unistd::unlinkat) to indicate that the path to remove is a
```

And, our `unlinkat()` function does not take a `AtFlags` but a [`UnlinkatFlags`](https://github.com/nix-rust/nix/blob/bfcb2269d1f5e97723a1639cf6a33a85814afdf3/src/unistd.rs#L1312-L1317).

File: src/fcntl.rs [hunk 3]
```diff
@@ -239,16 +273,30 @@ pub fn renameat<P1: ?Sized + NixPath, P2: ?Sized + NixPath>(
...
+        /// This operation makes sense only for overlay/union filesystem implementations.
         RENAME_WHITEOUT;
     }
 }
 
 feature! {
 #![feature = "fs"]
+/// Like [`renameat`], but with an additional `flags` argument.
+///
+/// A `renameat2` call with a zero flags argument is equivalent to `renameat`.
```
Comment (SteveLauC) [inline src/fcntl.rs:296]: Maybe the following one is more suitable for our interface:
```suggestion
/// A `renameat2` call with an empty flags argument is equivalent to `renameat`.
```

File: src/fcntl.rs [hunk 4]
```diff
@@ -584,17 +679,27 @@ pub fn fcntl(fd: RawFd, arg: FcntlArg) -> Result<c_int> {
     Errno::result(res)
 }
 
+/// Operations for use with [`flock`].
```
Comment (SteveLauC) [inline src/fcntl.rs:682]: The `flock()` function is deprecated so let's use the new `Flock::lock()` interface:
```suggestion
/// Operations for use with [`Flock::lock()`].
```

File: src/fcntl.rs [hunk 5]
```diff
@@ -584,17 +679,27 @@ pub fn fcntl(fd: RawFd, arg: FcntlArg) -> Result<c_int> {
...
+#[cfg(not(any(target_os = "redox", target_os = "solaris")))]
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
 #[non_exhaustive]
 pub enum FlockArg {
+    /// shared file lock
     LockShared,
+    /// exclusive file lock
     LockExclusive,
+    /// Unlock file
     Unlock,
```
Comment (SteveLauC) [inline src/fcntl.rs:692]: Maybe we can also deprecate the `Unlock` variant given that this `FlockArg` type is only used with the lock interface: `Flock::lock()`

File: src/time.rs [hunk 6]
```diff
@@ -1,3 +1,4 @@
+//! Sleep, query system clocks, and set system closk
```
Comment (SteveLauC) [inline src/time.rs:1]: Seems to be a typo:

```suggestion
//! Sleep, query system clocks, and set system clock
```

File: src/time.rs [hunk 7]
```diff
@@ -57,20 +57,30 @@ impl ClockId {
     }
 
     #[cfg(any(linux_android, target_os = "emscripten", target_os = "fuchsia"))]
+    /// Starts at zero when the kernel boots and increments monotonically in SI seconds while the
+    /// machine is running.
     pub const CLOCK_BOOTTIME: ClockId = ClockId(libc::CLOCK_BOOTTIME);
+    /// Like [`CLOCK_BOOTTIME`](ClockId::CLOCK_BOOTTIME), but will wake the  system  if  it  is
```
Comment (SteveLauC) [inline src/time.rs:63]:
```suggestion
    /// Like [`CLOCK_BOOTTIME`](ClockId::CLOCK_BOOTTIME), but will wake the system if it is
```

File: src/time.rs [hunk 8]
```diff
@@ -57,20 +57,30 @@ impl ClockId {
...
     #[cfg(freebsdlike)]
+    /// Like [`CLOCK_MONOTONIC`](ClockId::CLOCK_MONOTONIC), but optimized for execution time at the expense of accuracy.
     pub const CLOCK_MONOTONIC_FAST: ClockId =
         ClockId(libc::CLOCK_MONOTONIC_FAST);
     #[cfg(freebsdlike)]
+    /// Like [`CLOCK_MONOTONIC`](ClockId::CLOCK_MONOTONIC), but optimized for accuracy at the expense of execution time.
     pub const CLOCK_MONOTONIC_PRECISE: ClockId =
         ClockId(libc::CLOCK_MONOTONIC_PRECISE);
+    /// Similar to [`CLOCK_MONOTONIC`](ClockId::CLOCK_MONOTONIC), but provides access to a raw
+    /// hardware-based time that  is  not subject  to  NTP adjustments.
```
Comment (SteveLauC) [inline src/time.rs:83]:
```suggestion
    /// hardware-based time that is not subject to NTP adjustments.
```

File: src/time.rs [hunk 9]
```diff
@@ -81,30 +91,43 @@ impl ClockId {
...
+    #[allow(missing_docs)] // Undocumented on Linux!
     #[cfg(any(
         target_os = "emscripten",
         target_os = "fuchsia",
         all(target_os = "linux", target_env = "musl")
     ))]
     pub const CLOCK_SGI_CYCLE: ClockId = ClockId(libc::CLOCK_SGI_CYCLE);
+    /// International Atomic Time.
+    ///
+    /// A  nonsettable  system-wide  clock derived from wall-clock time but ignoring leap seconds.
```
Comment (SteveLauC) [inline src/time.rs:130]:
```suggestion
    /// A nonsettable system-wide clock derived from wall-clock time but ignoring leap seconds.
```

**Review by SteveLauC — CHANGES_REQUESTED** [review]

File: src/fcntl.rs (hunk 2 above)
Comment (asomers) [reply src/fcntl.rs:52]: In that case, this flag is totally unused.  I'll mark it as hidden and deprecated.

**Review by SteveLauC — APPROVED** [review]

**Pull request merged**